src/cdk/a11y/input-modality/input-modality-detector.ts

```diff
@@ -119,17 +120,22 @@ export class InputModalityDetector implements OnDestroy {
   }
 
   /**
-   * Handles mouse events. Must be an arrow function in order to preserve the context when it gets
-   * bound.
+   * Handles mousedown events. Must be an arrow function in order to preserve the context when it
+   * gets bound.
    */
   private _onMousedown = (event: MouseEvent) => {
-    if (isFakeMousedownFromScreenReader(event)) { return; }
-
     // Touches trigger both touch and mouse events, so we need to distinguish between mouse events
     // that were triggered via mouse vs touch. To do so, check if the mouse event occurs closely
     // after the previous touch event.
     if (Date.now() - this._lastTouchMs < TOUCH_BUFFER_MS) { return; }
 
+    // Fake mousedown events are fired by some screen readers when controls are activated by the
+    // screen reader. Attribute this to keyboard input modality.
+    if (isFakeMousedownFromScreenReader(event)) {
```

Comment:
My understanding is that screen readers emit fake mouse events in these scenarios because web developers will sometimes hide important behavior behind mouse event listeners instead of click handlers.

> Could this be dispatched by assistive technology that isn't keyboard-based?

I've only ever seen these fake events dispatched when ATs are used with keyboard. The common example is performing an element's default action with a screen reader with a particular keystroke (on VO, CTRL + OPTION + Space). A less common example is using your keyboard to control your mouse (on MacOS, you can use mouse keys to control your mouse via keyboard, in which fake mouse events are emitted on click). I'm unfortunately not super familiar with other AT software beyond screen readers (I know of other AT devices such as switch controls, joysticks, head pointers, and eye tracking devices, but my understanding is that they are means to control one's mouse/keyboard, and don't interface directly with the device).

I guess I'll note that attributing these fake events to keyboard _is_ `FocusMonitor`'s current behavior, so this is not a change in behavior.

Reply:
Yeah, this seems niche enough that I wouldn't worry about it unless it comes up in a report